src/tree.rs

```diff
@@ -331,24 +331,15 @@ impl<T> Tree<T> {
             return Result::Err(error.expect("Tree::remove_node_lift_children: Missing an error value on finding an invalid NodeId."));
         }
 
-        if !self.node_has_parent(&node_id) {
-            self.clear_parent_of_children(&node_id);
-        } else {
-            let parent_id = self.get_unsafe(&node_id)
-                .parent()
-                .expect("Tree::remove_node_lift_children: node_has_parent() is true, but unwrapping said parent failed.")
-                .clone();
-
-            let children = self.get_unsafe(&node_id)
-                .children()
-                .clone();
-
-            for child_id in children {
+        if let Some(parent_id) = self.get_unsafe(&node_id).parent().cloned() {
+            //attach children to parent
+            for child_id in self.get_unsafe(&node_id).children().clone() {
                 self.set_as_parent_and_child(&parent_id, &child_id);
             }
+        } else {
+            self.clear_parent_of_children(&node_id);
```

Comment:
So, I went back and forth over this for a while back when I first wrote it.

I don't like this behavior to be honest, but for whatever reason (I can't seem to remember why) I decided that this was the best way to go a that time.

Sadly, changing this would be another breaking change as this behavior is documented this way, so some people might be depending on it.  In addition, to return an `Error` I would need a new type of `Error` to return.

I'm not against changing this behavior, but it would have to wait until the bump to 1.0.0 I think.